src/Symfony/Component/Form/Extension/Core/Type/DateTimeType.php

```diff
@@ -278,6 +278,28 @@ public function configureOptions(OptionsResolver $resolver)
             'text',
             'choice',
         ));
+
+        $resolver->setNormalizer('date_format', function (Options $options, $dateFormat) {
+            if (null !== $dateFormat && 'single_text' === $options['widget']) {
+                @trigger_error(sprintf('Using the "date_format" option of the %s when the "widget" option is set to "single_text" is deprecated since Symfony 4.2.', self::class), E_USER_DEPRECATED);
+            }
+
+            return $dateFormat;
+        });
+        $resolver->setNormalizer('date_widget', function (Options $options, $dateWidget) {
+            if (null !== $dateWidget && 'single_text' === $options['widget']) {
+                @trigger_error(sprintf('Using the "date_widget" option of the %s when the "widget" option is set to "single_text" is deprecated since Symfony 4.2.', self::class), E_USER_DEPRECATED);
+            }
+
+            return $dateWidget;
+        });
+        $resolver->setNormalizer('time_widget', function (Options $options, $timeWidget) {
```

Comment:
@xabbuh what do you think about to pass a second argument `Options $options` to the [closure value](https://github.com/symfony/symfony/blob/b1ae30586876925527a8bcf4b62aa7b10399b04b/src/Symfony/Component/OptionsResolver/OptionsResolver.php#L863) in `setDeprecated()` method for this kind of need? I mean, any ppl could be overriding this normalizer and the deprecation msg wouldn't show.

Somthing like this maybe:
```php
$resolver->setDeprecated('widget', function ($widget, Options $options) {
    if ('single_text' === $widget) {
        if (null !== $options['date_format']) {
            // @trigger_error ...
        }

        // ...
    }

    return '';
});
```
thus we guarantee that the deprecation message will always be displayed. Even, we're in time to change the arguments position `(Options $options, $value)` to be consistent with other closure signatures.

Reply:
I could work on it if you think it's a good idea.

Reply:
`$resolver->setDeprecated` is about deprecating options, not about deprecating specific values of the option currently.

Reply:
@stof you can, see https://github.com/symfony/symfony/pull/27277 and it's documented in https://github.com/symfony/symfony-docs/pull/9859

Reply:
@yceruto Sounds like it could be useful. Would you like to work on the changes for the OptionsResolver component?

Reply:
See proposal https://github.com/symfony/symfony/pull/28738

Reply:
works quite nice with your changes from #28738 👍